src/library.js

```diff
@@ -2323,6 +2323,19 @@ LibraryManager.library = {
                                "_emscripten_get_now = () => performance.now();\n",
 #endif
 
+  emscripten_get_timezone__sig: 'iv',
+  emscripten_get_timezone: function() {
+    var buflen = 128;
+
+    if (!_emscripten_get_timezone.buffer) {
+      _emscripten_get_timezone.buffer = _malloc(buflen);
```

Comment:
Let avoid the malloc dependency by having emscripten_get_timezone return a pointer to static memory. 

To do that you will want to create `emscripten_get_timezone` as a C function that contain `static char buf[128];`. then then have that call out to emscripten_get_timezone_js() passing the target pointer as arg0.

emscripten_get_timezone_js can then call emscriptenGetTimeZone() which returns a JS string.

Then you can use `emscriptenGetTimeZone` below too to avoid the extre UTF8ToString in wasi.js.

The declaration of `emscriptenGetTimeZone` should be prefixed with `$` to mark it as a JS-only function

Reply:
@sbc100 I am having trouble putting together what you are describing can you point me to a pattern of this elsewhere ? I was following `emscripten_get_window_title`

Reply:
Looks like to figured it out!